src/lib/cloud-provider.js

```diff
@@ -163,6 +183,10 @@ class CloudProvider {
         this.vm = null;
         this.username = null;
         this.projectId = null;
+        if (this._connectionTimeout) {
+            clearTimeout(this._connectionTimeout);
+            this._connectionTimeout = null;
+        }
```

Comment:
Do we need to reset connectAttempts here?

Reply:
Yes